cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -375,9 +382,63 @@ func (o *Options) Run(ctx context.Context) error {
 		return err
 	}
 
+	if err := server.GenericAPIServer.AddPostStartHook("bootstrap-public-info", func(_ genericapiserver.PostStartHookContext) error {
+		if err := createNamespaces(gardencorev1beta1.GardenerSystemPublicNamespace); err != nil {
+			return err
+		}
+
+		p := publicInfo{
+			Version:      version.Get().String(),
+			FeatureGates: fmt.Sprint(features.DefaultFeatureGate),
+		}
+
+		if len(o.ExtraOptions.WorkloadIdentityTokenIssuer) != 0 {
+			p.WorkloadIdentityIssuerURL = &o.ExtraOptions.WorkloadIdentityTokenIssuer
+		}
+
+		marshelledInfo, err := yaml.Marshal(p)
+		if err != nil {
+			return err
+		}
+
+		configMapName := "gardener-info"
+		gardenerAPIServerKey := "gardenerAPIServer"
+
+		configMap, err := kubeClient.CoreV1().ConfigMaps(gardencorev1beta1.GardenerSystemPublicNamespace).Get(ctx, configMapName, metav1.GetOptions{})
+		if err != nil {
+			if client.IgnoreNotFound(err) != nil {
+				return err
+			}
+
+			configMap := corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: gardencorev1beta1.GardenerSystemPublicNamespace,
+					Name:      configMapName,
+				},
+				Data: map[string]string{
+					gardenerAPIServerKey: string(marshelledInfo),
+				},
+			}
+			_, err = kubeClient.CoreV1().ConfigMaps(gardencorev1beta1.GardenerSystemPublicNamespace).Create(ctx, &configMap, metav1.CreateOptions{})
+			return err
+		}
+
+		configMap.Data[gardenerAPIServerKey] = string(marshelledInfo)
+		_, err = kubeClient.CoreV1().ConfigMaps(gardencorev1beta1.GardenerSystemPublicNamespace).Update(ctx, configMap, metav1.UpdateOptions{})
+		return err
+	}); err != nil {
+		return err
+	}
+
 	return server.GenericAPIServer.PrepareRun().Run(ctx.Done())
 }
 
+type publicInfo struct {
+	Version                   string  `json:"version" yaml:"version"`
+	WorkloadIdentityIssuerURL *string `json:"workloadIdentityIssuerURL,omitempty" yaml:"workloadIdentityIssuerURL,omitempty"`
```

Comment:
Interesting fact: Once go 1.24 is released we would be able to use `omitzero` and the pointer would not be needed https://tip.golang.org/doc/go1.24